pilot/pkg/networking/core/v1alpha3/loadbalancer/loadbalancer.go

```diff
@@ -110,8 +110,10 @@ func applyLocalityWeight(
 				// the load balancing weight for a locality is divided by the sum of the weights of all localities
 				for index, originalWeight := range destLocMap {
 					weight := float64(originalWeight*weight) / float64(totalWeight)
-					loadAssignment.Endpoints[index].LoadBalancingWeight = &wrappers.UInt32Value{
-						Value: uint32(math.Ceil(weight)),
+					if weight > 0 {
```

Comment:
How can weight=0?

Reply:
TBH, I have n't looked at the complete logic to see if it is possible  or not - But as far as I have searched in the code, this is the only place where a calculated value is set - So I thought it is safe to add that check. But If you know for sure, this is not going to be zero, I will close this PR and search to see if any thing else is problem.

Reply:
I am not pretty sure, but just find an error: the two `weight` is misused.

Reply:
Looking at it more closely, you may be right - It is not possible to be zero. So closing this

Reply:
Can you fix the error?
>weight := float64(originalWeight*weight) ...

Reply:
the first weight is a new var

Reply:
Good catch. Fixed. PTAL.